compiler/rustc_sanitizers/src/lib.rs

```diff
@@ -0,0 +1,7 @@
+#![feature(let_chains)]
+/// Sanitizers support for the Rust compiler.
+///
+/// This crate contains the source code for providing support for the sanitizers to the Rust
+/// compiler.
+pub mod cfi;
+pub mod kcfi;
```

Comment:
nit: we could probably simplify the file structure a little bit here, it's mostly an aesthetic concern so only if you agree:

```
rustc_sanitizers
|- lib.rs
|- cfi.rs (containing what is in cfi/typeid/mod.rs)
|- kcfi.rs (containing what is in kcfi/typeid/mod.rs)
 \- typeid/* (private mod, containing what is in cfi/typeid/itanium_cxx_abi now)
```

It's also possible that this doesn't make sense anticipating some of the other things you want to move to this crate that exists elsewhere at the moment, but if we're only anticipating things that we might do by having this structure (e.g. manglings other than itanium), then we can change to this structure when we're adding that.

Reply:
Yes, I thought about it too. The reason I ended up doing it this way was because I'm anticipating two things:

1. Move as much as possible of CFI and KCFI related source code to cfi/mod.rs and kcfi/mod.rs (e.g., source code at declare.rs and builder.rs as cfi::emit_type_test and kcfi::emit_kcfi_bundle).
2. Add possibly at least an additional encoding for Microsoft XFG and cross-language Microsoft XFG (but we might want to discuss if the rustc_sanitizers crate is the right place for it), or a more coarse-grained encoding for cross-language CFI compatibility with other languages.

What do you think?